![Tinder](https://full-stack-assets.s3.eu-west-3.amazonaws.com/M03-EDA/Tinder-Symbole.png)

# 🧹 Etape : Nettoyage des Données Speed Dating

## 📋 Objectifs du notebook

Ce notebook va :
1. Charger les données brutes
2. Renommer les colonnes en français selon la documentation
3. Faire une premiere analyse
4. Garder uniquement les colonnes utiles pour notre analyse
5. Créer un fichier propre pour l'analyse

---

## ⚠️ IMPORTANT

Les données ont été collectées lors de 21 sessions différentes (waves).

Le système de notation a changé entre certaines vagues !
- **Waves 1-5** : notation 100 points à distribuer
- **Waves 6-9** : échelle 1-10 (⚠️ PROBLÈME - échelle différente !)
- **Waves 10-21** : retour au système 100 points

le systeme de notation sur la vague 12 est différentes avec une contrainte, il faut supprimer la vague pour ne pas avoir de probleme dans l'analyse.


la question finales est ce qu il y a un autre rdv ? on va s'arreter sur les infos du premier temps et puis les données ensuites. et surtout pourquoi il match...


---
## 🔋 1. Chargement des librairies

In [478]:
import pandas as pd
import numpy as np

print('✅ librairies chargées !')

✅ librairies chargées !


---
## 💽 2. Import du dataset

In [479]:
# Charger les données brutes
df = pd.read_csv('Speed+Dating+Data.csv', encoding='latin-1')
df.head().sort_values(by='iid')

iid   id  gender  idg  condtn  wave  round  position  positin1  order  ...  \
0    1  1.0       0    1       1     1     10         7       NaN      4  ...   
1    1  1.0       0    1       1     1     10         7       NaN      3  ...   
2    1  1.0       0    1       1     1     10         7       NaN     10  ...   
3    1  1.0       0    1       1     1     10         7       NaN      5  ...   
4    1  1.0       0    1       1     1     10         7       NaN      7  ...   

   attr3_3  sinc3_3  intel3_3  fun3_3  amb3_3  attr5_3  sinc5_3  intel5_3  \
0      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
1      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
2      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
3      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
4      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   

   fun5_3  amb5_3  
0     NaN     NaN  
1     NaN     NaN  
2     NaN     NaN  
3     NaN     NaN  
4     NaN     NaN  

[5 rows x 195 columns]

In [480]:
#analyse rapide du dataset
print('le nombre de lignes est de :', df.shape[0])
print('le nombre de colonnes est de :', df.shape[1])
print('le nombre de participants uniques est de :', df["iid"].nunique())
print('le nombre de waves est de :', df["wave"].nunique())

le nombre de lignes est de : 8378
le nombre de colonnes est de : 195
le nombre de participants uniques est de : 551
le nombre de waves est de : 21


In [481]:
# visualisation des colonnes et des types de données
print(df.dtypes)
print(df.columns)
# on ne va pas voir ttes les colonnes
print(df.columns.tolist())


iid           int64
id          float64
gender        int64
idg           int64
condtn        int64
             ...   
attr5_3     float64
sinc5_3     float64
intel5_3    float64
fun5_3      float64
amb5_3      float64
Length: 195, dtype: object
Index(['iid', 'id', 'gender', 'idg', 'condtn', 'wave', 'round', 'position',
       'positin1', 'order',
       ...
       'attr3_3', 'sinc3_3', 'intel3_3', 'fun3_3', 'amb3_3', 'attr5_3',
       'sinc5_3', 'intel5_3', 'fun5_3', 'amb5_3'],
      dtype='object', length=195)
['iid', 'id', 'gender', 'idg', 'condtn', 'wave', 'round', 'position', 'positin1', 'order', 'partner', 'pid', 'match', 'int_corr', 'samerace', 'age_o', 'race_o', 'pf_o_att', 'pf_o_sin', 'pf_o_int', 'pf_o_fun', 'pf_o_amb', 'pf_o_sha', 'dec_o', 'attr_o', 'sinc_o', 'intel_o', 'fun_o', 'amb_o', 'shar_o', 'like_o', 'prob_o', 'met_o', 'age', 'field', 'field_cd', 'undergra', 'mn_sat', 'tuition', 'race', 'imprace', 'imprelig', 'from', 'zipcode', 'income', 'goal', 'date', 'go_out', 'car

## Trop de colonnes ...


In [482]:
# j'ai besoin de visualiser rapidement les valeurs uniques de certaines colonnes
valeur = 'attr1_1'
print('les valeurs unique de la colonnes', df[valeur].unique())

les valeurs unique de la colonnes [ 15.    45.    35.    20.    10.     9.09  60.    19.    30.    50.
  33.33 100.    12.    17.      nan  25.    40.     5.    21.    14.
  18.    80.    16.67  12.77   6.67  18.18  17.39  18.75  18.6   20.83
  14.29   8.33  16.    11.36  15.09   8.51  14.55   9.76  15.38  19.05
  15.91  15.22  20.51  27.78  11.11  17.78  16.28  19.57  18.37   9.52
  12.24  13.04  19.15  16.07  16.98  19.44  15.56  20.45  20.93  17.65
  17.5   13.21  19.61  14.58  17.31  11.54  16.36  15.52  17.02  14.89
   7.5   21.28  17.24  14.71  25.64  13.51  21.43  23.81  90.    22.
   0.     7.    24.    27.    23.     2.     9.    31.58  75.     8.
  70.    95.    55.    28.    58.  ]


---
## 🏷️ 3. Renommage des Colonnes en Français

Basé sur le fichier **Speed Dating Data Key.docx** que j'ai traduit avec Word.

In [483]:
# Dictionnaire complet de renommage des colonnes que nous allons garder.
rename_dict = {
    # identifiants
    'iid': 'id_unique',
    'id': 'id_wave',
    'gender': 'genre',
    'idg': 'id_genre',
    'wave': 'vague',
    'round': 'nb_personnes_vague',
    'order': 'ordre_rencontre',
    'partner': 'id_partenaire',
    'pid': 'id_unique_partenaire', 
    'match': 'match',
    
    # informations sur le partenaire
    'age_o': 'age_partenaire',
    'race_o': 'race_partenaire',
    'dec_o': 'decision_partenaire',
    'int_corr': 'correlation_interets',
    'samerace': 'same_race',
    'met_o': 'deja_rencontre_partenaire',
    
    # note du partenaire note reçue
    'attr_o': 'attractivite_recue',
    'sinc_o': 'sincerite_recue',
    'intel_o': 'intelligence_recue',
    'fun_o': 'fun_recu',
    'amb_o': 'ambition_recue',
    'shar_o': 'interets_partages_recus',
    'like_o': 'appreciation_recue',
    'prob_o': 'probabilite_oui_recue',
    
    # preferences du partenaire
    'pf_o_att': 'pref_partenaire_attractivite',
    'pf_o_sin': 'pref_partenaire_sincerite',
    'pf_o_int': 'pref_partenaire_intelligence',
    'pf_o_fun': 'pref_partenaire_fun',
    'pf_o_amb': 'pref_partenaire_ambition',
    'pf_o_sha': 'pref_partenaire_interets',
    
    # note données au partenaire
    'dec': 'decision',
    'attr': 'attractivite_donnee',
    'sinc': 'sincerite_donnee',
    'intel': 'intelligence_donnee',
    'fun': 'fun_donne',
    'amb': 'ambition_donnee',
    'shar': 'interets_partages_donnes',
    'like': 'appreciation_donnee',
    'prob': 'probabilite_oui_donnee',
    'met': 'deja_rencontre',
    
    # informations personnelles
    'age': 'age',
    'field': 'domaine_etude',
    'field_cd': 'domaine_code',
    'race': 'race',
    'imprace': 'importance_meme_race',
    'imprelig': 'importance_meme_religion',
    'goal': 'objectif_participation',
    'date': 'frequence_dates',
    'go_out': 'frequence_sorties',
    'career': 'carriere_visee',
    'career_c': 'carriere_code',
    
    # hobbies et intérêts
    'sports': 'interet_sport',
    'tvsports': 'interet_sport_tv',
    'exercise': 'interet_exercice',
    'dining': 'interet_restaurant',
    'museums': 'interet_musees',
    'art': 'interet_art',
    'hiking': 'interet_randonnee',
    'gaming': 'interet_jeux',
    'clubbing': 'interet_boite_nuit',
    'reading': 'interet_lecture',
    'tv': 'interet_tv',
    'theater': 'interet_theatre',
    'movies': 'interet_cinema',
    'concerts': 'interet_concerts',
    'music': 'interet_musique',
    'shopping': 'interet_shopping',
    'yoga': 'interet_yoga',
    
    # atente
    'exphappy': 'bonheur_attendu',
    
    # préférences persos
    'attr1_1': 'pref_attractivite',
    'sinc1_1': 'pref_sincerite',
    'intel1_1': 'pref_intelligence',
    'fun1_1': 'pref_fun',
    'amb1_1': 'pref_ambition',
    'shar1_1': 'pref_interets',
    
    # auto-évaluation
    'attr3_1': 'auto_attractivite',
    'sinc3_1': 'auto_sincerite',
    'intel3_1': 'auto_intelligence',
    'fun3_1': 'auto_fun',
    'amb3_1': 'auto_ambition',
}

print(f'Dictionnaire avec {len(rename_dict)} colonnes à renommer.')

Dictionnaire avec 80 colonnes à renommer.


In [484]:
# Renommer les colonnes et on va garder uniquement celles dans le dictionnaire
df = df.rename(columns=rename_dict)
df = df[list(rename_dict.values())]
df.head()

id_unique  id_wave  genre  id_genre  vague  nb_personnes_vague  \
0          1      1.0      0         1      1                  10   
1          1      1.0      0         1      1                  10   
2          1      1.0      0         1      1                  10   
3          1      1.0      0         1      1                  10   
4          1      1.0      0         1      1                  10   

   ordre_rencontre  id_partenaire  id_unique_partenaire  match  ...  \
0                4              1                  11.0      0  ...   
1                3              2                  12.0      0  ...   
2               10              3                  13.0      1  ...   
3                5              4                  14.0      1  ...   
4                7              5                  15.0      1  ...   

   pref_sincerite  pref_intelligence  pref_fun  pref_ambition  pref_interets  \
0            20.0               20.0      15.0           15.0           15.0   
1            20.0               20.0      15.0           15.0           15.0   
2            20.0               20.0      15.0           15.0           15.0   
3            20.0               20.0      15.0           15.0           15.0   
4            20.0               20.0      15.0           15.0           15.0   

   auto_attractivite  auto_sincerite  auto_intelligence  auto_fun  \
0                6.0             8.0                8.0       8.0   
1                6.0             8.0                8.0       8.0   
2                6.0             8.0                8.0       8.0   
3                6.0             8.0                8.0       8.0   
4                6.0             8.0                8.0       8.0   

   auto_ambition  
0            7.0  
1            7.0  
2            7.0  
3            7.0  
4            7.0  

[5 rows x 80 columns]

---
## 🔍 4. Analyse des Valeurs Manquantes

In [485]:
# Analyse des valeurs manquantes

missing_values = df.isnull().sum()
missing_values = missing_values[missing_values > 0].sort_values(ascending=False)
missing_values = pd.DataFrame(missing_values)
missing_values.columns = ['nombre_valeurs_manquantes']
missing_values['pourcentage_valeurs_manquantes'] = (missing_values['nombre_valeurs_manquantes'] / df.shape[0]) * 100
missing_values.head(10)

nombre_valeurs_manquantes  \
interets_partages_recus                         1076   
interets_partages_donnes                        1067   
ambition_recue                                   722   
ambition_donnee                                  712   
deja_rencontre_partenaire                        385   
deja_rencontre                                   375   
fun_recu                                         360   
fun_donne                                        350   
probabilite_oui_recue                            318   
probabilite_oui_donnee                           309   

                           pourcentage_valeurs_manquantes  
interets_partages_recus                         12.843161  
interets_partages_donnes                        12.735736  
ambition_recue                                   8.617809  
ambition_donnee                                  8.498448  
deja_rencontre_partenaire                        4.595369  
deja_rencontre                                   4.476009  
fun_recu                                         4.296968  
fun_donne                                        4.177608  
probabilite_oui_recue                            3.795655  
probabilite_oui_donnee                           3.688231

In [486]:
#supprimer la vague 12
df = df[df['vague'] != 12]
df['vague'].unique()

array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 13, 14, 15, 16, 17, 18,
       19, 20, 21])

In [487]:
#pid en NAN on les supprime
df = df.dropna(subset=['id_unique_partenaire'])
df['id_unique_partenaire'].isna().sum()
#verification reprise du code de distribution
print('le nombre de lignes est de :', df.shape[0])
print('le nombre de colonnes est de :', df.shape[1])
print('le nombre de participants uniques est de :', df["id_unique"].nunique())
print('le nombre de waves est de :', df["vague"].nunique())

le nombre de lignes est de : 7976
le nombre de colonnes est de : 80
le nombre de participants uniques est de : 523
le nombre de waves est de : 20


---
## 📊 Nettoyage des données.

In [488]:
# Aperçu
df.head(10)
df.describe(include='all')

id_unique      id_wave        genre     id_genre        vague  \
count   7976.000000  7975.000000  7976.000000  7976.000000  7976.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     282.702106     9.025831     0.500000    17.474674    11.326981   
std      162.337072     5.538731     0.500031    11.046274     6.139315   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      148.000000     4.000000     0.000000     8.000000     7.000000   
50%      272.000000     8.000000     0.500000    16.000000    11.000000   
75%      413.000000    13.000000     1.000000    26.000000    15.000000   
max      552.000000    22.000000     1.000000    44.000000    21.000000   

        nb_personnes_vague  ordre_rencontre  id_partenaire  \
count          7976.000000      7976.000000    7976.000000   
unique                 NaN              NaN            NaN   
top                    NaN              NaN            NaN   
freq                   NaN              NaN            NaN   
mean             17.021815         9.002131       9.027457   
std               4.412098         5.529935       5.540288   
min               5.000000         1.000000       1.000000   
25%              15.000000         4.000000       4.000000   
50%              18.000000         8.000000       8.000000   
75%              20.000000        13.000000      13.000000   
max              22.000000        22.000000      22.000000   

        id_unique_partenaire        match  ...  pref_sincerite  \
count            7976.000000  7976.000000  ...     7897.000000   
unique                   NaN          NaN  ...             NaN   
top                      NaN          NaN  ...             NaN   
freq                     NaN          NaN  ...             NaN   
mean              282.702106     0.167753  ...       17.272880   
std               162.337072     0.373670  ...        6.882485   
min                 1.000000     0.000000  ...        0.000000   
25%               148.000000     0.000000  ...       15.000000   
50%               272.000000     0.000000  ...       18.000000   
75%               413.000000     0.000000  ...       20.000000   
max               552.000000     1.000000  ...       60.000000   

        pref_intelligence     pref_fun  pref_ambition  pref_interets  \
count         7897.000000  7888.000000    7879.000000    7857.000000   
unique                NaN          NaN            NaN            NaN   
top                   NaN          NaN            NaN            NaN   
freq                  NaN          NaN            NaN            NaN   
mean            20.231263    17.559730      10.713629      11.840703   
std              6.735399     6.117449       6.176301       6.439463   
min              0.000000     0.000000       0.000000       0.000000   
25%             17.500000    15.000000       5.000000       8.330000   
50%             20.000000    18.000000      10.000000      11.110000   
75%             23.260000    20.000000      15.000000      16.000000   
max             50.000000    50.000000      53.000000      30.000000   

        auto_attractivite  auto_sincerite  auto_intelligence     auto_fun  \
count         7871.000000     7871.000000        7871.000000  7871.000000   
unique                NaN             NaN                NaN          NaN   
top                   NaN             NaN                NaN          NaN   
freq                  NaN             NaN                NaN          NaN   
mean             7.083217        8.276712           8.399695     7.693177   
std              1.408113        1.399007           1.088729     1.581662   
min              2.000000        2.000000           3.000000     2.000000   
25%              6.000000        8.000000           8.000000     7.000000   
50%          

In [489]:
# Informations
print('il y a tjs un nombre de participant unique :', df["id_unique"].nunique())
print('il y a tjs un nombre de vague :', df["vague"].nunique())
print('nous avons tjs des valeurs manquantes :', df.isnull().sum().sum())

il y a tjs un nombre de participant unique : 523
il y a tjs un nombre de vague : 20
nous avons tjs des valeurs manquantes : 11775


In [490]:
# Statistiques descriptives
df.describe(include='all')

id_unique      id_wave        genre     id_genre        vague  \
count   7976.000000  7975.000000  7976.000000  7976.000000  7976.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     282.702106     9.025831     0.500000    17.474674    11.326981   
std      162.337072     5.538731     0.500031    11.046274     6.139315   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      148.000000     4.000000     0.000000     8.000000     7.000000   
50%      272.000000     8.000000     0.500000    16.000000    11.000000   
75%      413.000000    13.000000     1.000000    26.000000    15.000000   
max      552.000000    22.000000     1.000000    44.000000    21.000000   

        nb_personnes_vague  ordre_rencontre  id_partenaire  \
count          7976.000000      7976.000000    7976.000000   
unique                 NaN              NaN            NaN   
top                    NaN              NaN            NaN   
freq                   NaN              NaN            NaN   
mean             17.021815         9.002131       9.027457   
std               4.412098         5.529935       5.540288   
min               5.000000         1.000000       1.000000   
25%              15.000000         4.000000       4.000000   
50%              18.000000         8.000000       8.000000   
75%              20.000000        13.000000      13.000000   
max              22.000000        22.000000      22.000000   

        id_unique_partenaire        match  ...  pref_sincerite  \
count            7976.000000  7976.000000  ...     7897.000000   
unique                   NaN          NaN  ...             NaN   
top                      NaN          NaN  ...             NaN   
freq                     NaN          NaN  ...             NaN   
mean              282.702106     0.167753  ...       17.272880   
std               162.337072     0.373670  ...        6.882485   
min                 1.000000     0.000000  ...        0.000000   
25%               148.000000     0.000000  ...       15.000000   
50%               272.000000     0.000000  ...       18.000000   
75%               413.000000     0.000000  ...       20.000000   
max               552.000000     1.000000  ...       60.000000   

        pref_intelligence     pref_fun  pref_ambition  pref_interets  \
count         7897.000000  7888.000000    7879.000000    7857.000000   
unique                NaN          NaN            NaN            NaN   
top                   NaN          NaN            NaN            NaN   
freq                  NaN          NaN            NaN            NaN   
mean            20.231263    17.559730      10.713629      11.840703   
std              6.735399     6.117449       6.176301       6.439463   
min              0.000000     0.000000       0.000000       0.000000   
25%             17.500000    15.000000       5.000000       8.330000   
50%             20.000000    18.000000      10.000000      11.110000   
75%             23.260000    20.000000      15.000000      16.000000   
max             50.000000    50.000000      53.000000      30.000000   

        auto_attractivite  auto_sincerite  auto_intelligence     auto_fun  \
count         7871.000000     7871.000000        7871.000000  7871.000000   
unique                NaN             NaN                NaN          NaN   
top                   NaN             NaN                NaN          NaN   
freq                  NaN             NaN                NaN          NaN   
mean             7.083217        8.276712           8.399695     7.693177   
std              1.408113        1.399007           1.088729     1.581662   
min              2.000000        2.000000           3.000000     2.000000   
25%              6.000000        8.000000           8.000000     7.000000   
50%          

In [491]:
# remplacement des valeurs manquantes pour les colonnes numériques par la médiane on va iterer sur chaque colonne avec boucle for et type int ou float. si il y a des valeurs superieurs a 10
numerical_cols = df.select_dtypes(exclude=['object']).columns.tolist()
for col in numerical_cols:
    median_value = df[col].median()
    df[col].fillna(median_value, inplace=True)
    

print('✅ Valeurs manquantes traitées !')

✅ Valeurs manquantes traitées !


/var/folders/h5/tjbnq0312nnbccsycqjy_ftc0000gn/T/ipykernel_64638/2427204173.py:5: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df[col].fillna(median_value, inplace=True)


In [492]:
# controle des valeurs manquantes, on reprend le code précédent

missing_values = df.isnull().sum()
missing_values = missing_values[missing_values > 0].sort_values(ascending=False)
missing_values = pd.DataFrame(missing_values)
missing_values.columns = ['nombre_valeurs_manquantes']
missing_values['pourcentage_valeurs_manquantes'] = (missing_values['nombre_valeurs_manquantes'] / df.shape[0]) * 100
missing_values.head(10)

nombre_valeurs_manquantes  pourcentage_valeurs_manquantes
carriere_visee                         89                        1.115848
domaine_etude                          63                        0.789870

In [493]:
df.describe(include='all')

id_unique      id_wave        genre     id_genre        vague  \
count   7976.000000  7976.000000  7976.000000  7976.000000  7976.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     282.702106     9.025702     0.500000    17.474674    11.326981   
std      162.337072     5.538395     0.500031    11.046274     6.139315   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      148.000000     4.000000     0.000000     8.000000     7.000000   
50%      272.000000     8.000000     0.500000    16.000000    11.000000   
75%      413.000000    13.000000     1.000000    26.000000    15.000000   
max      552.000000    22.000000     1.000000    44.000000    21.000000   

        nb_personnes_vague  ordre_rencontre  id_partenaire  \
count          7976.000000      7976.000000    7976.000000   
unique                 NaN              NaN            NaN   
top                    NaN              NaN            NaN   
freq                   NaN              NaN            NaN   
mean             17.021815         9.002131       9.027457   
std               4.412098         5.529935       5.540288   
min               5.000000         1.000000       1.000000   
25%              15.000000         4.000000       4.000000   
50%              18.000000         8.000000       8.000000   
75%              20.000000        13.000000      13.000000   
max              22.000000        22.000000      22.000000   

        id_unique_partenaire        match  ...  pref_sincerite  \
count            7976.000000  7976.000000  ...     7976.000000   
unique                   NaN          NaN  ...             NaN   
top                      NaN          NaN  ...             NaN   
freq                     NaN          NaN  ...             NaN   
mean              282.702106     0.167753  ...       17.280081   
std               162.337072     0.373670  ...        6.848690   
min                 1.000000     0.000000  ...        0.000000   
25%               148.000000     0.000000  ...       15.000000   
50%               272.000000     0.000000  ...       18.000000   
75%               413.000000     0.000000  ...       20.000000   
max               552.000000     1.000000  ...       60.000000   

        pref_intelligence     pref_fun  pref_ambition  pref_interets  \
count         7976.000000  7976.000000    7976.000000    7976.000000   
unique                NaN          NaN            NaN            NaN   
top                   NaN          NaN            NaN            NaN   
freq                  NaN          NaN            NaN            NaN   
mean            20.228972    17.564588      10.704950      11.829801   
std              6.701995     6.083778       6.139123       6.391853   
min              0.000000     0.000000       0.000000       0.000000   
25%             17.650000    15.000000       5.000000       9.090000   
50%             20.000000    18.000000      10.000000      11.110000   
75%             23.260000    20.000000      15.000000      15.690000   
max             50.000000    50.000000      53.000000      30.000000   

        auto_attractivite  auto_sincerite  auto_intelligence     auto_fun  \
count         7976.000000     7976.000000        7976.000000  7976.000000   
unique                NaN             NaN                NaN          NaN   
top                   NaN             NaN                NaN          NaN   
freq                  NaN             NaN                NaN          NaN   
mean             7.082121        8.273069           8.394433     7.697217   
std              1.398845        1.390124           1.082498     1.571605   
min              2.000000        2.000000           3.000000     2.000000   
25%              6.000000        8.000000           8.000000     7.000000   
50%          

voila pour le nettoyage ! Je me dis que j'aurais déja du nettoyer certaines colonnes sur les vagues 6-9 car elles ont une notation différente.


---
## 📋 7. Vérification des vagues (échelles différentes)

In [494]:
#on va regarder la moyennes des notes données et reçues sur les participants pour vague 6-9
vagues_a_verifier = [6, 7, 8, 9]
colonnes_notes = [
    'pref_attractivite', 'pref_sincerite', 'pref_intelligence', 'pref_fun', 'pref_ambition', 'pref_interets',
    'pref_partenaire_attractivite', 'pref_partenaire_sincerite', 'pref_partenaire_intelligence',
    'pref_partenaire_fun', 'pref_partenaire_ambition', 'pref_partenaire_interets'
]
for vague in vagues_a_verifier:
    df_vague = df[df['vague'] == vague]
    print(f'Vague {vague} :')
    for col in colonnes_notes:
        mean_value = df_vague[col].mean()
        print(f'  Moyenne de {col} : {mean_value:.2f}')
    print('les notes sont comprises sont une distribution de 0 a 100')



Vague 6 :
  Moyenne de pref_attractivite : 16.99
  Moyenne de pref_sincerite : 18.77
  Moyenne de pref_intelligence : 18.11
  Moyenne de pref_fun : 17.66
  Moyenne de pref_ambition : 12.53
  Moyenne de pref_interets : 15.66
  Moyenne de pref_partenaire_attractivite : 16.99
  Moyenne de pref_partenaire_sincerite : 18.77
  Moyenne de pref_partenaire_intelligence : 18.11
  Moyenne de pref_partenaire_fun : 17.66
  Moyenne de pref_partenaire_ambition : 12.53
  Moyenne de pref_partenaire_interets : 15.66
les notes sont comprises sont une distribution de 0 a 100
Vague 7 :
  Moyenne de pref_attractivite : 15.57
  Moyenne de pref_sincerite : 17.94
  Moyenne de pref_intelligence : 19.16
  Moyenne de pref_fun : 18.44
  Moyenne de pref_ambition : 15.21
  Moyenne de pref_interets : 13.68
  Moyenne de pref_partenaire_attractivite : 15.57
  Moyenne de pref_partenaire_sincerite : 17.94
  Moyenne de pref_partenaire_intelligence : 19.16
  Moyenne de pref_partenaire_fun : 18.44
  Moyenne de pref_partenai

In [495]:
#on va controler la moyenne des notes données et reçues sur les participants pour vague 1-5
vagues_a_verifier = [1, 2, 3, 4, 5,10,11, 13,14,15,16,17,18,19,20,21]
colonnes_notes = [
    'pref_attractivite', 'pref_sincerite', 'pref_intelligence', 'pref_fun', 'pref_ambition', 'pref_interets',
    'pref_partenaire_attractivite', 'pref_partenaire_sincerite', 'pref_partenaire_intelligence',
    'pref_partenaire_fun', 'pref_partenaire_ambition', 'pref_partenaire_interets'
]
for vague in vagues_a_verifier:
    df_vague = df[df['vague'] == vague]
    print(f'Vague {vague} :')
    for col in colonnes_notes:
        mean_value = df_vague[col].mean()
        print(f'  Moyenne de {col} : {mean_value:.2f}')
    print('les notes sont comprises sont une distribution de 0 a 100')



Vague 1 :
  Moyenne de pref_attractivite : 32.32
  Moyenne de pref_sincerite : 11.11
  Moyenne de pref_intelligence : 19.87
  Moyenne de pref_fun : 19.61
  Moyenne de pref_ambition : 7.41
  Moyenne de pref_interets : 9.67
  Moyenne de pref_partenaire_attractivite : 32.32
  Moyenne de pref_partenaire_sincerite : 11.11
  Moyenne de pref_partenaire_intelligence : 19.87
  Moyenne de pref_partenaire_fun : 19.61
  Moyenne de pref_partenaire_ambition : 7.41
  Moyenne de pref_partenaire_interets : 9.67
les notes sont comprises sont une distribution de 0 a 100
Vague 2 :
  Moyenne de pref_attractivite : 24.13
  Moyenne de pref_sincerite : 18.18
  Moyenne de pref_intelligence : 23.20
  Moyenne de pref_fun : 16.75
  Moyenne de pref_ambition : 9.08
  Moyenne de pref_interets : 9.22
  Moyenne de pref_partenaire_attractivite : 24.13
  Moyenne de pref_partenaire_sincerite : 18.18
  Moyenne de pref_partenaire_intelligence : 23.20
  Moyenne de pref_partenaire_fun : 16.75
  Moyenne de pref_partenaire_amb

In [496]:
'''# on va mettre a l'echelle les notes sur une attribution de 100 points pour les vagues 6-9
# Méthode : (valeur / somme_par_ligne) * 100

total_recues= ['attractivite_recue', 'sincerite_recue', 'intelligence_recue', 'fun_recu', 'ambition_recue', 'interets_partages_recus']
total_donnees = ['attractivite_donnee', 'sincerite_donnee', 'intelligence_donnee', 'fun_donne', 'ambition_donnee', 'interets_partages_donnes']

# Créer un masque pour les vagues 6-9
mask_waves_69 = df['vague'].between(6, 9)

# Calculer les sommes par ligne UNIQUEMENT pour les vagues 6-9
somme_recues = df.loc[mask_waves_69, total_recues].sum(axis=1)
somme_donnees = df.loc[mask_waves_69, total_donnees].sum(axis=1)

# Normaliser chaque colonne
for col in total_recues:
    mask_valid = (somme_recues > 0) & df.loc[mask_waves_69, col].notna()
    df.loc[mask_waves_69 & mask_valid, col] = (
        df.loc[mask_waves_69 & mask_valid, col] / somme_recues[mask_valid] * 100
    )

for col in total_donnees:
    mask_valid = (somme_donnees > 0) & df.loc[mask_waves_69, col].notna()
    df.loc[mask_waves_69 & mask_valid, col] = (
        df.loc[mask_waves_69 & mask_valid, col] / somme_donnees[mask_valid] * 100
    )'''

"# on va mettre a l'echelle les notes sur une attribution de 100 points pour les vagues 6-9\n# Méthode : (valeur / somme_par_ligne) * 100\n\ntotal_recues= ['attractivite_recue', 'sincerite_recue', 'intelligence_recue', 'fun_recu', 'ambition_recue', 'interets_partages_recus']\ntotal_donnees = ['attractivite_donnee', 'sincerite_donnee', 'intelligence_donnee', 'fun_donne', 'ambition_donnee', 'interets_partages_donnes']\n\n# Créer un masque pour les vagues 6-9\nmask_waves_69 = df['vague'].between(6, 9)\n\n# Calculer les sommes par ligne UNIQUEMENT pour les vagues 6-9\nsomme_recues = df.loc[mask_waves_69, total_recues].sum(axis=1)\nsomme_donnees = df.loc[mask_waves_69, total_donnees].sum(axis=1)\n\n# Normaliser chaque colonne\nfor col in total_recues:\n    mask_valid = (somme_recues > 0) & df.loc[mask_waves_69, col].notna()\n    df.loc[mask_waves_69 & mask_valid, col] = (\n        df.loc[mask_waves_69 & mask_valid, col] / somme_recues[mask_valid] * 100\n    )\n\nfor col in total_donnees:\n 

In [497]:
'''# on controle les moyennes des colonnes normalisées (vagues 6-9 seulement)
df_vagues_69 = df[df['vague'].between(6, 9)]
print('Statistiques pour les vagues 6-9 après normalisation :')
df_vagues_69[total_recues + total_donnees].describe()'''

"# on controle les moyennes des colonnes normalisées (vagues 6-9 seulement)\ndf_vagues_69 = df[df['vague'].between(6, 9)]\nprint('Statistiques pour les vagues 6-9 après normalisation :')\ndf_vagues_69[total_recues + total_donnees].describe()"

In [498]:
'''# Vérifier que les sommes par ligne font bien 100 pour les vagues 6-9
df_vagues_69 = df[df['vague'].between(6, 9)]
somme_finale_recues = df_vagues_69[total_recues].sum(axis=1)
somme_finale_donnees = df_vagues_69[total_donnees].sum(axis=1)

print('✅ Normalisation terminée !')
print(f'Somme moyenne des notes reçues (vagues 6-9) : {somme_finale_recues.mean():.2f}')
print(f'Somme moyenne des notes données (vagues 6-9) : {somme_finale_donnees.mean():.2f}')'''

"# Vérifier que les sommes par ligne font bien 100 pour les vagues 6-9\ndf_vagues_69 = df[df['vague'].between(6, 9)]\nsomme_finale_recues = df_vagues_69[total_recues].sum(axis=1)\nsomme_finale_donnees = df_vagues_69[total_donnees].sum(axis=1)\n\nprint('✅ Normalisation terminée !')\nprint(f'Somme moyenne des notes reçues (vagues 6-9) : {somme_finale_recues.mean():.2f}')\nprint(f'Somme moyenne des notes données (vagues 6-9) : {somme_finale_donnees.mean():.2f}')"

In [499]:
'''# dernier describe pour controle sur les colonnes modifiées
df[total_recues + total_donnees].describe(include='all')
'''

"# dernier describe pour controle sur les colonnes modifiées\ndf[total_recues + total_donnees].describe(include='all')\n"

---
## 🖥️ 8. renommer les values pour avoir qq chose de propre

In [500]:
# remplacer sur la colonne gender 1 par homme et 0 par femme
df['genre'] = df['genre'].replace({1: 'homme', 0: 'femme'})
# remplacer sur la colonne match 1 par oui et 0 par non
df['match'] = df['match'].replace({1: 'oui', 0: 'non'})
# remplacer sur la colonne deja_rencontre 1 par oui et 0 par non
df['deja_rencontre'] = df['deja_rencontre'].replace({1: 'oui', 0: 'non'})


# preparation de la colonne carriere code
career = {  1: 'Avocat',
            2: 'Universitaire/Recherche',
            3: 'Psychologue',
            4: 'Médecin/Médecine',
            5: 'Ingénieur',
            6: 'Arts créatifs/divertissement',
            7: 'Banque/Conseil/Finance/Marketing/Business/PDG/Entrepreneur/Admin',
            8: 'Immobilier',    
            9: 'Affaires internationales/humanitaires',
            10: 'Indécis',
            11: 'Travail social',
            12: 'Orthophonie',
            13: 'Politique',
            14: 'Sports professionnels/Athlétisme',
            15: 'Autre',
            16: 'Journalisme',
            17: 'Architecture'
}

df['carriere_code'] = df['carriere_code'].replace(career)

# preparation de la colonne domaine_code 
domaine = {  1: 'Loi',
            2: 'Mathématiques',
            3: 'Sciences sociales, Psychologue',
            4: 'Sciences médicales, produits pharmaceutiques et biotechnologie',
            5: 'Ingénierie',
            6: 'Anglais/Écriture créative/ Journalisme',
            7: 'Histoire/Religion/Philosophie',
            8: 'Affaires/Économie/Finance',
            9: 'Éducation, Milieu universitaire',
            10: 'Sciences biologiques/Chimie/Physique',
            11: 'Travail social',
            12: 'Étudiant de premier cycle/indécis',
            13: 'Science politique/Affaires internationales',
            14: 'Pellicule',
            15: 'Beaux-arts/Administration des arts',
            16: 'Langues',
            17: 'Architecture',
            18: 'Autre'
}

df['domaine_code'] = df['domaine_code'].replace(domaine)

# preparation de la colonne objectif_participation
objectif = {  1: 'Ça avait l\'air d\'être une soirée amusante',
              2: 'Pour rencontrer de nouvelles personnes',
              3: 'Pour obtenir une date',
              4: 'À la recherche d\'une relation sérieuse',
              5: 'Dire que je l\'ai fait',
              6: 'Autre'
}

df['objectif_participation'] = df['objectif_participation'].replace(objectif)





In [501]:
df.head()

id_unique  id_wave  genre  id_genre  vague  nb_personnes_vague  \
0          1      1.0  femme         1      1                  10   
1          1      1.0  femme         1      1                  10   
2          1      1.0  femme         1      1                  10   
3          1      1.0  femme         1      1                  10   
4          1      1.0  femme         1      1                  10   

   ordre_rencontre  id_partenaire  id_unique_partenaire match  ...  \
0                4              1                  11.0   non  ...   
1                3              2                  12.0   non  ...   
2               10              3                  13.0   oui  ...   
3                5              4                  14.0   oui  ...   
4                7              5                  15.0   oui  ...   

   pref_sincerite  pref_intelligence  pref_fun  pref_ambition  pref_interets  \
0            20.0               20.0      15.0           15.0           15.0   
1            20.0               20.0      15.0           15.0           15.0   
2            20.0               20.0      15.0           15.0           15.0   
3            20.0               20.0      15.0           15.0           15.0   
4            20.0               20.0      15.0           15.0           15.0   

   auto_attractivite  auto_sincerite  auto_intelligence  auto_fun  \
0                6.0             8.0                8.0       8.0   
1                6.0             8.0                8.0       8.0   
2                6.0             8.0                8.0       8.0   
3                6.0             8.0                8.0       8.0   
4                6.0             8.0                8.0       8.0   

   auto_ambition  
0            7.0  
1            7.0  
2            7.0  
3            7.0  
4            7.0  

[5 rows x 80 columns]

## ➡️ 9. Insertion de la colonne regroupement age

In [503]:
# decoupage par tranche d'age et par sexe cf lien https://www.delftstack.com/fr/api/python-pandas/pandas-cut-function/
tranches_age = [18, 25, 35, 45, 60]
labels_age = ['18-24', '25-34', '35-44', '45-59']
df['tranche_age'] = pd.cut(df['age'], bins=tranches_age, labels=labels_age, right=False)



## 💾 10. Sauvegarde du Dataset Nettoyé

In [504]:
# Sauvegarder le dataset nettoyé avec colonnes renommées
output_file = '../speed_dating_clean.csv'
df.to_csv(output_file, index=False, encoding='utf-8-sig')
print('fichier sauvegardé !')


fichier sauvegardé !


---
## ✅ Résumé du Nettoyage

### Ce qui a été fait :
1. ✅ **Chargement** : 8,378 lignes × 195 colonnes
2. ✅ **Renommage et conservation des données interressantes pour notre analyse** : Toutes les colonnes en français selon la documentation traduite avec word
2. ✅ **suppression vague 12** : système de notation différent et non compatible
3. ✅ **Remplacement des valeurs manquantes** : mediane pour remplacer les valeurs manquantes
4. ✅ **description des values** : indication des valeurs a place de code dans certains champs
5. ✅ **Sauvegarde** : Dataset nettoyé exporté
5. ✅ **le dataset clean** : 7976 lignes × 80 colonnes et 523 participants sur 20 vagues.

